diff --git a/library/functions.sh b/library/functions.sh
index 675ecc1..3460318 100644
--- a/library/functions.sh
+++ b/library/functions.sh
@@ -438,7 +438,6 @@ function func_download {
 					--tries=$_WGET_TRIES \
 					--timeout=$_WGET_TIMEOUT \
 					--wait=$_WGET_WAIT \
-					--no-check-certificate \
 					$_url -O $_lib_name > $_log_name 2>&1
 				_result=$?
 			}
diff --git a/scripts/clang-3.3.sh b/scripts/clang-3.3.sh
index f203ce1..05d1ca4 100644
--- a/scripts/clang-3.3.sh
+++ b/scripts/clang-3.3.sh
@@ -40,9 +40,9 @@ PKG_NAME=llvm-${PKG_VERSION}.src
 PKG_DIR_NAME=llvm-${PKG_VERSION}.src
 PKG_TYPE=.tar.gz
 PKG_URLS=(
-	"http://llvm.org/releases/${PKG_VERSION}/llvm-${PKG_VERSION}.src${PKG_TYPE}"
-	"http://llvm.org/releases/${PKG_VERSION}/cfe-${PKG_VERSION}.src${PKG_TYPE}|dir:$PKG_NAME/tools"
-	"http://llvm.org/releases/${PKG_VERSION}/compiler-rt-${PKG_VERSION}.src${PKG_TYPE}|dir:$PKG_NAME/projects"
+	"https://llvm.org/releases/${PKG_VERSION}/llvm-${PKG_VERSION}.src${PKG_TYPE}"
+	"https://llvm.org/releases/${PKG_VERSION}/cfe-${PKG_VERSION}.src${PKG_TYPE}|dir:$PKG_NAME/tools"
+	"https://llvm.org/releases/${PKG_VERSION}/compiler-rt-${PKG_VERSION}.src${PKG_TYPE}|dir:$PKG_NAME/projects"
 )
 
 #
diff --git a/scripts/clang-3.4.sh b/scripts/clang-3.4.sh
index 9b08b79..547a3cc 100644
--- a/scripts/clang-3.4.sh
+++ b/scripts/clang-3.4.sh
@@ -40,9 +40,9 @@ PKG_NAME=llvm-${PKG_VERSION}
 PKG_DIR_NAME=llvm-${PKG_VERSION}
 PKG_TYPE=.tar.gz
 PKG_URLS=(
-	"http://llvm.org/releases/${PKG_VERSION}/llvm-${PKG_VERSION}.src${PKG_TYPE}"
-	"http://llvm.org/releases/${PKG_VERSION}/clang-${PKG_VERSION}.src${PKG_TYPE}|dir:$PKG_NAME/tools"
-	"http://llvm.org/releases/${PKG_VERSION}/compiler-rt-${PKG_VERSION}.src${PKG_TYPE}|dir:$PKG_NAME/projects"
+	"https://llvm.org/releases/${PKG_VERSION}/llvm-${PKG_VERSION}.src${PKG_TYPE}"
+	"https://llvm.org/releases/${PKG_VERSION}/clang-${PKG_VERSION}.src${PKG_TYPE}|dir:$PKG_NAME/tools"
+	"https://llvm.org/releases/${PKG_VERSION}/compiler-rt-${PKG_VERSION}.src${PKG_TYPE}|dir:$PKG_NAME/projects"
 )
 
 #
diff --git a/scripts/clang-git.sh b/scripts/clang-git.sh
index 18172c9..bb6137c 100644
--- a/scripts/clang-git.sh
+++ b/scripts/clang-git.sh
@@ -40,10 +40,10 @@ PKG_DIR_NAME=llvm-git
 PKG_TYPE=git
 
 PKG_URLS=(
-	"http://llvm.org/git/llvm.git|repo:$PKG_TYPE|module:$PKG_NAME"
-	"http://llvm.org/git/clang.git|repo:$PKG_TYPE|dir:$PKG_NAME/tools|module:clang"
-	"http://llvm.org/git/compiler-rt.git|repo:$PKG_TYPE|dir:$PKG_NAME/projects|module:compiler-rt"
-	"http://llvm.org/git/test-suite.git|repo:$PKG_TYPE|dir:$PKG_NAME/projects|module:test-suite"
+	"https://llvm.org/git/llvm.git|repo:$PKG_TYPE|module:$PKG_NAME"
+	"https://llvm.org/git/clang.git|repo:$PKG_TYPE|dir:$PKG_NAME/tools|module:clang"
+	"https://llvm.org/git/compiler-rt.git|repo:$PKG_TYPE|dir:$PKG_NAME/projects|module:compiler-rt"
+	"https://llvm.org/git/test-suite.git|repo:$PKG_TYPE|dir:$PKG_NAME/projects|module:test-suite"
 )
 
 PKG_PRIORITY=main
diff --git a/scripts/cloog.sh b/scripts/cloog.sh
index a5a759a..2937598 100644
--- a/scripts/cloog.sh
+++ b/scripts/cloog.sh
@@ -40,7 +40,7 @@ PKG_NAME=$BUILD_ARCHITECTURE-cloog-${PKG_VERSION}-$LINK_TYPE_SUFFIX
 PKG_DIR_NAME=cloog-${PKG_VERSION}
 PKG_TYPE=.tar.gz
 PKG_URLS=(
-	"http://www.bastoul.net/cloog/pages/download/cloog-${PKG_VERSION}${PKG_TYPE}"
+	"https://www.bastoul.net/cloog/pages/download/cloog-${PKG_VERSION}${PKG_TYPE}"
 )
 PKG_PRIORITY=prereq
 
diff --git a/scripts/gcc-10-branch.sh b/scripts/gcc-10-branch.sh
index 742a7fb..b4f1f8a 100644
--- a/scripts/gcc-10-branch.sh
+++ b/scripts/gcc-10-branch.sh
@@ -40,7 +40,7 @@ PKG_NAME=gcc-${PKG_VERSION}-branch
 PKG_DIR_NAME=gcc-${PKG_VERSION}-branch
 PKG_TYPE=git
 PKG_URLS=(
-	"git://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
+	"https://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
 )
 
 PKG_PRIORITY=main
diff --git a/scripts/gcc-11-branch.sh b/scripts/gcc-11-branch.sh
index 53ec980..6986b05 100644
--- a/scripts/gcc-11-branch.sh
+++ b/scripts/gcc-11-branch.sh
@@ -40,7 +40,7 @@ PKG_NAME=gcc-${PKG_VERSION}-branch
 PKG_DIR_NAME=gcc-${PKG_VERSION}-branch
 PKG_TYPE=git
 PKG_URLS=(
-	"git://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
+	"https://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
 )
 
 PKG_PRIORITY=main
diff --git a/scripts/gcc-4.6-branch.sh b/scripts/gcc-4.6-branch.sh
index 0dd2962..bd17bef 100644
--- a/scripts/gcc-4.6-branch.sh
+++ b/scripts/gcc-4.6-branch.sh
@@ -40,7 +40,7 @@ PKG_NAME=gcc-${PKG_VERSION}-branch
 PKG_DIR_NAME=gcc-${PKG_VERSION}-branch
 PKG_TYPE=git
 PKG_URLS=(
-	"git://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
+	"https://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
 )
 
 PKG_PRIORITY=main
diff --git a/scripts/gcc-4.7-branch.sh b/scripts/gcc-4.7-branch.sh
index ddff22c..52a5211 100644
--- a/scripts/gcc-4.7-branch.sh
+++ b/scripts/gcc-4.7-branch.sh
@@ -40,7 +40,7 @@ PKG_NAME=gcc-${PKG_VERSION}-branch
 PKG_DIR_NAME=gcc-${PKG_VERSION}-branch
 PKG_TYPE=git
 PKG_URLS=(
-	"git://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
+	"https://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
 )
 
 PKG_PRIORITY=main
diff --git a/scripts/gcc-4.8-branch.sh b/scripts/gcc-4.8-branch.sh
index 4438d89..b1f0395 100644
--- a/scripts/gcc-4.8-branch.sh
+++ b/scripts/gcc-4.8-branch.sh
@@ -40,7 +40,7 @@ PKG_NAME=gcc-${PKG_VERSION}-branch
 PKG_DIR_NAME=gcc-${PKG_VERSION}-branch
 PKG_TYPE=git
 PKG_URLS=(
-	"git://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
+	"https://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
 )
 
 PKG_PRIORITY=main
diff --git a/scripts/gcc-4.9-branch.sh b/scripts/gcc-4.9-branch.sh
index f13851d..2470bec 100644
--- a/scripts/gcc-4.9-branch.sh
+++ b/scripts/gcc-4.9-branch.sh
@@ -40,7 +40,7 @@ PKG_NAME=gcc-${PKG_VERSION}-branch
 PKG_DIR_NAME=gcc-${PKG_VERSION}-branch
 PKG_TYPE=git
 PKG_URLS=(
-	"git://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
+	"https://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
 )
 
 PKG_PRIORITY=main
diff --git a/scripts/gcc-5-branch.sh b/scripts/gcc-5-branch.sh
index 86bc762..99948f5 100644
--- a/scripts/gcc-5-branch.sh
+++ b/scripts/gcc-5-branch.sh
@@ -40,7 +40,7 @@ PKG_NAME=gcc-${PKG_VERSION}-branch
 PKG_DIR_NAME=gcc-${PKG_VERSION}-branch
 PKG_TYPE=git
 PKG_URLS=(
-	"git://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
+	"https://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
 )
 
 PKG_PRIORITY=main
diff --git a/scripts/gcc-6-branch.sh b/scripts/gcc-6-branch.sh
index cdb3568..847238f 100644
--- a/scripts/gcc-6-branch.sh
+++ b/scripts/gcc-6-branch.sh
@@ -40,7 +40,7 @@ PKG_NAME=gcc-${PKG_VERSION}-branch
 PKG_DIR_NAME=gcc-${PKG_VERSION}-branch
 PKG_TYPE=git
 PKG_URLS=(
-	"git://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
+	"https://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
 )
 
 PKG_PRIORITY=main
diff --git a/scripts/gcc-7-branch.sh b/scripts/gcc-7-branch.sh
index 587ce06..a4de60c 100644
--- a/scripts/gcc-7-branch.sh
+++ b/scripts/gcc-7-branch.sh
@@ -40,7 +40,7 @@ PKG_NAME=gcc-${PKG_VERSION}-branch
 PKG_DIR_NAME=gcc-${PKG_VERSION}-branch
 PKG_TYPE=git
 PKG_URLS=(
-	"git://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
+	"https://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
 )
 
 PKG_PRIORITY=main
diff --git a/scripts/gcc-8-branch.sh b/scripts/gcc-8-branch.sh
index 8596c61..6d19c2a 100644
--- a/scripts/gcc-8-branch.sh
+++ b/scripts/gcc-8-branch.sh
@@ -40,7 +40,7 @@ PKG_NAME=gcc-${PKG_VERSION}-branch
 PKG_DIR_NAME=gcc-${PKG_VERSION}-branch
 PKG_TYPE=git
 PKG_URLS=(
-	"git://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
+	"https://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
 )
 
 PKG_PRIORITY=main
diff --git a/scripts/gcc-9-branch.sh b/scripts/gcc-9-branch.sh
index 3650e57..430de25 100644
--- a/scripts/gcc-9-branch.sh
+++ b/scripts/gcc-9-branch.sh
@@ -40,7 +40,7 @@ PKG_NAME=gcc-${PKG_VERSION}-branch
 PKG_DIR_NAME=gcc-${PKG_VERSION}-branch
 PKG_TYPE=git
 PKG_URLS=(
-	"git://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
+	"https://gcc.gnu.org/git/gcc.git|branch:releases/gcc-$PKG_VERSION|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
 )
 
 PKG_PRIORITY=main
diff --git a/scripts/gcc-trunk.sh b/scripts/gcc-trunk.sh
index 296f71d..6768f91 100644
--- a/scripts/gcc-trunk.sh
+++ b/scripts/gcc-trunk.sh
@@ -40,7 +40,7 @@ PKG_NAME=gcc-${PKG_VERSION}
 PKG_DIR_NAME=gcc-${PKG_VERSION}
 PKG_TYPE=git
 PKG_URLS=(
-	"git://gcc.gnu.org/git/gcc.git|branch:master|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
+	"https://gcc.gnu.org/git/gcc.git|branch:master|repo:$PKG_TYPE|module:$PKG_DIR_NAME"
 )
 
 PKG_PRIORITY=main
diff --git a/scripts/libffi.sh b/scripts/libffi.sh
index 42905f8..5daff81 100644
--- a/scripts/libffi.sh
+++ b/scripts/libffi.sh
@@ -40,7 +40,7 @@ PKG_NAME=libffi-${PKG_VERSION}
 PKG_DIR_NAME=libffi-${PKG_VERSION}
 PKG_TYPE=.tar.gz
 PKG_URLS=(
-	"ftp://sourceware.org/pub/libffi/libffi-${PKG_VERSION}${PKG_TYPE}"
+	"https://sourceware.org/pub/libffi/libffi-${PKG_VERSION}${PKG_TYPE}"
 )
 
 PKG_PRIORITY=extra
diff --git a/scripts/make_git.sh b/scripts/make_git.sh
index 3653602..0ec0edd 100644
--- a/scripts/make_git.sh
+++ b/scripts/make_git.sh
@@ -39,7 +39,7 @@ PKG_NAME=make_git
 PKG_DIR_NAME=make_git
 PKG_TYPE=git
 PKG_URLS=(
-	"http://git.savannah.gnu.org/cgit/make.git|repo:$PKG_TYPE|module:$PKG_NAME"
+	"https://git.savannah.gnu.org/cgit/make.git|repo:$PKG_TYPE|module:$PKG_NAME"
 )
 
 PKG_PRIORITY=extra
diff --git a/scripts/make_git_bat.sh b/scripts/make_git_bat.sh
index 918755f..b63a2ec 100644
--- a/scripts/make_git_bat.sh
+++ b/scripts/make_git_bat.sh
@@ -39,7 +39,7 @@ PKG_NAME=make_git
 PKG_DIR_NAME=make_git
 PKG_TYPE=git
 PKG_URLS=(
-	"http://git.savannah.gnu.org/cgit/make.git|repo:$PKG_TYPE|module:$PKG_NAME"
+	"https://git.savannah.gnu.org/cgit/make.git|repo:$PKG_TYPE|module:$PKG_NAME"
 )
 
 PKG_PRIORITY=extra
diff --git a/scripts/mingw-w64-download.sh b/scripts/mingw-w64-download.sh
index 384542b..933e575 100644
--- a/scripts/mingw-w64-download.sh
+++ b/scripts/mingw-w64-download.sh
@@ -41,13 +41,13 @@ PKG_NAME=mingw-w64-${RUNTIME_VERSION}
 	MINGW_PKG_DIR_VERSION_SUFFIX="-${RUNTIME_VERSION}"
 	PKG_TYPE=.tar.bz2
 	PKG_URLS=(
-		"http://downloads.sourceforge.net/project/mingw-w64/mingw-w64/mingw-w64-release/mingw-w64-${RUNTIME_VERSION}${PKG_TYPE}"
+		"https://downloads.sourceforge.net/project/mingw-w64/mingw-w64/mingw-w64-release/mingw-w64-${RUNTIME_VERSION}${PKG_TYPE}"
 	)
 } || {
 	MINGW_PKG_DIR_VERSION_SUFFIX=""
 	PKG_TYPE=git
 	PKG_URLS=(
-		"git://git.code.sf.net/p/mingw-w64/mingw-w64|branch:$RUNTIME_BRANCH|repo:$PKG_TYPE"
+		"https://git.code.sf.net/p/mingw-w64/mingw-w64|branch:$RUNTIME_BRANCH|repo:$PKG_TYPE"
 	)
 }
 
diff --git a/scripts/mingw-w64-libraries-pseh.sh b/scripts/mingw-w64-libraries-pseh.sh
index 4449749..822a89e 100644
--- a/scripts/mingw-w64-libraries-pseh.sh
+++ b/scripts/mingw-w64-libraries-pseh.sh
@@ -40,7 +40,7 @@ PKG_DIR_NAME=pseh-${RUNTIME_VERSION}
 PKG_DIR_NAME=mingw-w64/mingw-w64-libraries/pseh
 PKG_TYPE=git
 PKG_URLS=(
-	"git://git.code.sf.net/p/mingw-w64/mingw-w64|branch:$RUNTIME_BRANCH|repo:$PKG_TYPE"
+	"https://git.code.sf.net/p/mingw-w64/mingw-w64|branch:$RUNTIME_BRANCH|repo:$PKG_TYPE"
 )
 
 PKG_PRIORITY=extra
diff --git a/scripts/ppl.sh b/scripts/ppl.sh
index b400a8b..5d89a6a 100644
--- a/scripts/ppl.sh
+++ b/scripts/ppl.sh
@@ -40,7 +40,7 @@ PKG_NAME=$BUILD_ARCHITECTURE-ppl-${PKG_VERSION}-$LINK_TYPE_SUFFIX
 PKG_DIR_NAME=ppl-${PKG_VERSION}
 PKG_TYPE=.tar.xz
 PKG_URLS=(
-	"http://bugseng.com/products/ppl/download/ftp/releases/${PKG_VERSION}/ppl-${PKG_VERSION}${PKG_TYPE}"
+	"https://bugseng.com/products/ppl/download/ftp/releases/${PKG_VERSION}/ppl-${PKG_VERSION}${PKG_TYPE}"
 )
 
 PKG_PRIORITY=prereq
diff --git a/scripts/python-2.sh b/scripts/python-2.sh
index 6c90d5c..4e88edb 100644
--- a/scripts/python-2.sh
+++ b/scripts/python-2.sh
@@ -40,7 +40,7 @@ PKG_NAME=Python-${PKG_VERSION}
 PKG_DIR_NAME=Python-${PKG_VERSION}
 PKG_TYPE=.tar.xz
 PKG_URLS=(
-	"http://www.python.org/ftp/python/${PKG_VERSION}/Python-${PKG_VERSION}${PKG_TYPE}"
+	"https://www.python.org/ftp/python/${PKG_VERSION}/Python-${PKG_VERSION}${PKG_TYPE}"
 )
 
 PKG_PRIORITY=extra
diff --git a/scripts/tcl.sh b/scripts/tcl.sh
index 651dbfa..af05ccc 100644
--- a/scripts/tcl.sh
+++ b/scripts/tcl.sh
@@ -41,7 +41,7 @@ PKG_DIR_NAME=tcl${PKG_VERSION}
 PKG_SUBDIR_NAME=win
 PKG_TYPE=.tar.gz
 PKG_URLS=(
-	"http://prdownloads.sourceforge.net/tcl/tcl${PKG_VERSION}-src${PKG_TYPE}"
+	"https://prdownloads.sourceforge.net/tcl/tcl${PKG_VERSION}-src${PKG_TYPE}"
 )
 
 PKG_PRIORITY=extra
diff --git a/scripts/tk.sh b/scripts/tk.sh
index 8d8896a..d8e7194 100644
--- a/scripts/tk.sh
+++ b/scripts/tk.sh
@@ -41,7 +41,7 @@ PKG_DIR_NAME=tk${PKG_VERSION}
 PKG_SUBDIR_NAME=win
 PKG_TYPE=.tar.gz
 PKG_URLS=(
-	"http://prdownloads.sourceforge.net/tcl/tk${PKG_VERSION}-src${PKG_TYPE}"
+	"https://prdownloads.sourceforge.net/tcl/tk${PKG_VERSION}-src${PKG_TYPE}"
 )
 
 PKG_PRIORITY=extra
diff --git a/scripts/xz-utils.sh b/scripts/xz-utils.sh
index 9b50d15..92bb22a 100644
--- a/scripts/xz-utils.sh
+++ b/scripts/xz-utils.sh
@@ -40,7 +40,7 @@ PKG_NAME=xz-${PKG_VERSION}
 PKG_DIR_NAME=xz-${PKG_VERSION}
 PKG_TYPE=.tar.xz
 PKG_URLS=(
-	"http://tukaani.org/xz/xz-${PKG_VERSION}${PKG_TYPE}"
+	"https://tukaani.org/xz/xz-${PKG_VERSION}${PKG_TYPE}"
 )
 
 PKG_PRIORITY=extra
