diff --git a/build b/build
index 40cb7ac..e3c6269 100755
--- a/build
+++ b/build
@@ -318,7 +318,7 @@ while [[ $# > 0 ]]; do
                     RUNTIME_BRANCH="v10.x"
                 ;;
 				trunk)
-					RUNTIME_VERSION=v9
+					RUNTIME_VERSION=v10
 					RUNTIME_BRANCH="master"
 				;;
 				v3.3.0)
diff --git a/library/config-win.sh b/library/config-win.sh
index 0192bff..11e2f43 100644
--- a/library/config-win.sh
+++ b/library/config-win.sh
@@ -101,6 +101,7 @@ function func_test_installed_packages {
 		texinfo
 		autogen
 		dejagnu
+		pkgconf
 	)
 
 	echo "--> installing required packages..."
diff --git a/library/config.sh b/library/config.sh
index 9e924c8..46e44ab 100644
--- a/library/config.sh
+++ b/library/config.sh
@@ -189,10 +189,10 @@ esac
 
 # **************************************************************************
 
-echo -n "-> Checking for installed packages... "
-func_test_installed_packages
-[[ $? == 1 ]] && exit 1
-echo "done"
+#echo -n "-> Checking for installed packages... "
+#func_test_installed_packages
+#[[ $? == 1 ]] && exit 1
+#echo "done"
 
 # **************************************************************************
 
diff --git a/library/functions.sh b/library/functions.sh
index 46e5f33..2e15f63 100644
--- a/library/functions.sh
+++ b/library/functions.sh
@@ -438,7 +438,6 @@ function func_download {
 					--tries=$_WGET_TRIES \
 					--timeout=$_WGET_TIMEOUT \
 					--wait=$_WGET_WAIT \
-					--no-check-certificate \
 					$_url -O $_lib_name > $_log_name 2>&1
 				_result=$?
 			}
diff --git a/scripts/bzip2.sh b/scripts/bzip2.sh
index a243eaa..bc6ba89 100644
--- a/scripts/bzip2.sh
+++ b/scripts/bzip2.sh
@@ -35,7 +35,7 @@
 
 # **************************************************************************
 
-PKG_VERSION=1.0.6
+PKG_VERSION=1.0.8
 PKG_NAME=bzip2-${PKG_VERSION}
 PKG_DIR_NAME=bzip2-${PKG_VERSION}
 PKG_TYPE=.tar.gz
@@ -50,10 +50,8 @@ PKG_PRIORITY=extra
 #
 
 PKG_PATCHES=(
-	bzip2/bzip2-1.0.4-bzip2recover.patch
 	bzip2/bzip2-1.0.6-autoconfiscated.patch
 	bzip2/bzip2-use-cdecl-calling-convention.patch
-	bzip2/bzip2-1.0.5-slash.patch
 )
 
 #
diff --git a/scripts/expat.sh b/scripts/expat.sh
index ed54957..8610139 100644
--- a/scripts/expat.sh
+++ b/scripts/expat.sh
@@ -35,7 +35,7 @@
 
 # **************************************************************************
 
-PKG_VERSION=2.4.7
+PKG_VERSION=2.4.9
 PKG_NAME=expat-${PKG_VERSION}
 PKG_DIR_NAME=expat-${PKG_VERSION}
 PKG_TYPE=.tar.xz
diff --git a/scripts/gcc-10.3.0.sh b/scripts/gcc-10.3.0.sh
index fd4b633..d8693bd 100644
--- a/scripts/gcc-10.3.0.sh
+++ b/scripts/gcc-10.3.0.sh
@@ -61,6 +61,9 @@ PKG_PATCHES=(
 	gcc/gcc-10-ktietz-libgomp.patch
 	gcc/gcc-libgomp-ftime64.patch
 	gcc/0020-libgomp-Don-t-hard-code-MS-printf-attributes.patch
+	$( [[ $RUNTIME_MAJOR_VERSION -ge 10 ]] \
+		&& echo "gcc/0001-gcc-config-i386-mingw32.h-Ensure-lmsvcrt-precede-lke.patch" \
+	)
 )
 
 #
diff --git a/scripts/libffi.sh b/scripts/libffi.sh
index 5daff81..24b9eaf 100644
--- a/scripts/libffi.sh
+++ b/scripts/libffi.sh
@@ -35,21 +35,19 @@
 
 # **************************************************************************
 
-PKG_VERSION=3.2.1
+PKG_VERSION=3.4.2
 PKG_NAME=libffi-${PKG_VERSION}
 PKG_DIR_NAME=libffi-${PKG_VERSION}
 PKG_TYPE=.tar.gz
 PKG_URLS=(
-	"https://sourceware.org/pub/libffi/libffi-${PKG_VERSION}${PKG_TYPE}"
+	"https://github.com/libffi/libffi/releases/download/v${PKG_VERSION}/libffi-${PKG_VERSION}${PKG_TYPE}"
 )
 
 PKG_PRIORITY=extra
 
 #
 
-PKG_PATCHES=(
-	libffi/fix_return_size.patch
-)
+PKG_PATCHES=()
 
 #
 
diff --git a/scripts/mingw-w64-crt.sh b/scripts/mingw-w64-crt.sh
index 4a8d6ea..08871b0 100644
--- a/scripts/mingw-w64-crt.sh
+++ b/scripts/mingw-w64-crt.sh
@@ -94,7 +94,8 @@ PKG_CONFIGURE_FLAGS=(
 #
 
 PKG_MAKE_FLAGS=(
-	-j$JOBS
+	#-j$JOBS
+	-j1
 	all
 )
 
diff --git a/scripts/openssl.sh b/scripts/openssl.sh
index 2464b9c..dab41e8 100644
--- a/scripts/openssl.sh
+++ b/scripts/openssl.sh
@@ -35,7 +35,7 @@
 
 # **************************************************************************
 
-PKG_VERSION=1.1.1k
+PKG_VERSION=1.1.1q
 PKG_NAME=openssl-${PKG_VERSION}
 PKG_DIR_NAME=openssl-${PKG_VERSION}
 PKG_TYPE=.tar.gz
diff --git a/scripts/python-3.sh b/scripts/python-3.sh
index 3d93bc9..5a4c0c3 100644
--- a/scripts/python-3.sh
+++ b/scripts/python-3.sh
@@ -35,7 +35,7 @@
 
 # **************************************************************************
 
-PKG_VERSION=3.9.10
+PKG_VERSION=3.10.7
 PKG_NAME=Python-${PKG_VERSION}
 PKG_DIR_NAME=Python-${PKG_VERSION}
 PKG_TYPE=git
@@ -48,7 +48,7 @@ PKG_PRIORITY=extra
 #
 
 PKG_EXECUTE_AFTER_UNCOMPRESS=(
-	"git reset --hard 12d1cb5" # Reset to this commit hash for reproducible builds
+	"git reset --hard c00cb40fdcbec2eed6a03ddbb39dfb270ac39154" # Reset to this commit hash for reproducible builds
 )
 
 #
@@ -101,10 +101,11 @@ PKG_CONFIGURE_FLAGS=(
 	# --with-tzpath=$LIBS_DIR/share/zoneinfo
 	--enable-optimizations
 	#
-	LIBFFI_INCLUDEDIR="$LIBSW_DIR/lib/libffi-$LIBFFI_VERSION/include"
+	PKG_CONFIG_PATH="$PREREQ_DIR/$BUILD_ARCHITECTURE-zlib-$LINK_TYPE_SUFFIX/lib/pkgconfig:$LIBS_DIR/lib/pkgconfig:$PKG_CONFIG_PATH"
 	CFLAGS="\"$COMMON_CFLAGS -D__USE_MINGW_ANSI_STDIO=1\""
 	CPPFLAGS="\"$COMMON_CPPFLAGS $MY_CPPFLAGS\""
 	LDFLAGS="\"$COMMON_LDFLAGS -L$PREREQW_DIR/$BUILD_ARCHITECTURE-zlib-$LINK_TYPE_SUFFIX/lib -L$LIBSW_DIR/lib\""
+	LIBS="\"-lffi -lole32 -loleaut32 -luuid\""
 )
 
 #
diff --git a/scripts/readline.sh b/scripts/readline.sh
index 6f7e561..f3a0fb1 100644
--- a/scripts/readline.sh
+++ b/scripts/readline.sh
@@ -35,7 +35,7 @@
 
 # **************************************************************************
 
-PKG_VERSION=8.1
+PKG_VERSION=8.1.2
 PKG_NAME=readline-${PKG_VERSION}
 PKG_DIR_NAME=readline-${PKG_VERSION}
 PKG_TYPE=.tar.gz
@@ -48,7 +48,7 @@ PKG_PRIORITY=extra
 #
 
 PKG_PATCHES=(
-	readline/readline81-001.patch
+	#readline/readline81-001.patch
 	readline/Fixed-missing-POSIX_SIGNALS-check.patch
 )
 
diff --git a/scripts/sqlite.sh b/scripts/sqlite.sh
index 9a5fda0..aa7f23d 100644
--- a/scripts/sqlite.sh
+++ b/scripts/sqlite.sh
@@ -35,12 +35,12 @@
 
 # **************************************************************************
 
-PKG_VERSION=3350500
+PKG_VERSION=3380500
 PKG_NAME=sqlite-${PKG_VERSION}
 PKG_DIR_NAME=sqlite-autoconf-${PKG_VERSION}
 PKG_TYPE=.tar.gz
 PKG_URLS=(
-	"https://www.sqlite.org/2021/sqlite-autoconf-${PKG_VERSION}${PKG_TYPE}"
+	"https://www.sqlite.org/2022/sqlite-autoconf-${PKG_VERSION}${PKG_TYPE}"
 )
 PKG_PRIORITY=extra
 
diff --git a/scripts/tcl.sh b/scripts/tcl.sh
index af05ccc..d20f9b4 100644
--- a/scripts/tcl.sh
+++ b/scripts/tcl.sh
@@ -35,7 +35,7 @@
 
 # **************************************************************************
 
-PKG_VERSION=8.6.11
+PKG_VERSION=8.6.12
 PKG_NAME=tcl${PKG_VERSION}
 PKG_DIR_NAME=tcl${PKG_VERSION}
 PKG_SUBDIR_NAME=win
diff --git a/scripts/tests.sh b/scripts/tests.sh
index 7668844..3cddb8a 100644
--- a/scripts/tests.sh
+++ b/scripts/tests.sh
@@ -81,15 +81,15 @@ lasterror_test2_list=(
 	"lasterror_test2.cpp -o lasterror_test2.exe"
 )
 
-[[ $MSVCRT_VERSION == ucrt ]] && (
-	time_test_list=(
-		"time_test.c -lpthread -lucrt -o time_test.exe"
-	)
-) || (
+#[[ $MSVCRT_VERSION == ucrt ]] && (
+#	time_test_list=(
+#		"time_test.c -lpthread -lucrt -o time_test.exe"
+#	)
+#) || (
 	time_test_list=(
 		"time_test.c -lpthread -o time_test.exe"
 	)
-)
+#)
 
 [[ `echo $BUILD_VERSION | cut -d. -f1` == 4 && `echo $BUILD_VERSION | cut -d. -f2` -le 6 ]] && (
 	sleep_test_list=(
diff --git a/scripts/tk.sh b/scripts/tk.sh
index d8e7194..40eb3f4 100644
--- a/scripts/tk.sh
+++ b/scripts/tk.sh
@@ -35,7 +35,7 @@
 
 # **************************************************************************
 
-PKG_VERSION=8.6.11
+PKG_VERSION=8.6.12
 PKG_NAME=tk${PKG_VERSION}
 PKG_DIR_NAME=tk${PKG_VERSION}
 PKG_SUBDIR_NAME=win
diff --git a/scripts/winpthreads.sh b/scripts/winpthreads.sh
index baf7e28..65fff85 100644
--- a/scripts/winpthreads.sh
+++ b/scripts/winpthreads.sh
@@ -67,11 +67,11 @@ PKG_CONFIGURE_FLAGS=(
     # mingw-builds) with a first/initial bootstrap build of a toolchain with ucrt,
     # this linkage isn't needed then either. so assume the first ucrt mingw-build
     # toolchain would be built with a "--rev=0" arg.
-	$( [[ $MSVCRT_VERSION == ucrt ]] && \
-		[[ $BOOTSTRAPING == no ]] && \
-			[[ $RUNTIME_MAJOR_VERSION -ge 10 ]] && \
-                [[ $REV_NUM -eq 0 ]] && \
-                    echo "LIBS=\"-lucrtbase\"" )
+	#$( [[ $MSVCRT_VERSION == ucrt ]] && \
+	#	[[ $BOOTSTRAPING == no ]] && \
+	#		[[ $RUNTIME_MAJOR_VERSION -ge 10 ]] && \
+    #            [[ $REV_NUM -eq 0 ]] && \
+    #                echo "LIBS=\"-lucrtbase\"" )
 	CFLAGS="\"$COMMON_CFLAGS\""
 	CXXFLAGS="\"$COMMON_CXXFLAGS\""
 	CPPFLAGS="\"$COMMON_CPPFLAGS\""
