diff --git a/scripts/tests.sh b/scripts/tests.sh
index 49fc894..145bbf4 100644
--- a/scripts/tests.sh
+++ b/scripts/tests.sh
@@ -116,7 +116,7 @@ PKG_TESTS["pthread_test"]=pthread_test_list[@]
 [[ $THREADS_MODEL == posix ]] && { PKG_TESTS["stdthread_test"]=stdthread_test_list[@]; }
 PKG_TESTS["lasterror_test1"]=lasterror_test1_list[@]
 PKG_TESTS["lasterror_test2"]=lasterror_test2_list[@]
-PKG_TESTS["time_test"]=time_test_list[@]
+#PKG_TESTS["time_test"]=time_test_list[@]
 [[ $THREADS_MODEL == posix ]] && { PKG_TESTS["sleep_test"]=sleep_test_list[@]; }
 [[ `echo $BUILD_VERSION | cut -d. -f1` -ge 6 ]] && { PKG_TESTS["random_device"]=random_device_list[@]; }
 [[ `echo $BUILD_VERSION | cut -d. -f1` -ge 7 ]] && { PKG_TESTS["filesystem"]=filesystem_list[@]; }
