diff --git a/scripts/expat.sh b/scripts/expat.sh
index 978df77..583b634 100644
--- a/scripts/expat.sh
+++ b/scripts/expat.sh
@@ -35,12 +35,12 @@
 
 # **************************************************************************
 
-PKG_VERSION=2.4.1
+PKG_VERSION=2.4.5
 PKG_NAME=expat-${PKG_VERSION}
 PKG_DIR_NAME=expat-${PKG_VERSION}
 PKG_TYPE=.tar.xz
 PKG_URLS=(
-	"http://sourceforge.net/projects/expat/files/expat/${PKG_VERSION}/${PKG_NAME}${PKG_TYPE}"
+	"https://sourceforge.net/projects/expat/files/expat/${PKG_VERSION}/${PKG_NAME}${PKG_TYPE}"
 )
 
 PKG_PRIORITY=extra
@@ -48,7 +48,7 @@ PKG_PRIORITY=extra
 #
 
 PKG_PATCHES=(
-	expat/expat_mingw_tests_fix.patch
+	# expat/expat_mingw_tests_fix.patch
 )
 
 #
